scripts/check-peer-deps.js

```diff
@@ -0,0 +1,124 @@
+const path = require('path');
+const { promises: fs } = require('fs');
+const glob = require('util').promisify(require('glob'));
+const {
+  pluginExternals,
+  sharedExternals,
+} = require('@mongodb-js/webpack-config-compass');
+
+function getImportsFromSourceString(sourceString) {
+  const imports = [];
+  const regexp =
+    // A non-type import or a require statement
+    /(?:(?:^|\n)import\s(?!type)[\s\S]*?("|')(?<import>[^\1]+?)\1|require(?:\.resolve)?\(("|')(?<require>[^\3]+?)\3\))/g;
+  let match = regexp.exec(sourceString);
+  while (match !== null) {
+    imports.push(match.groups.import ?? match.groups.require);
+    match = regexp.exec(sourceString);
+  }
+  return Array.from(new Set(imports))
+    .map((name) => {
+      // Keep second part for namespaced imports, but drop everything else
+      const [a, b] = name.split('/');
+      return a.startsWith('@') ? `${a}/${b}` : a;
```

Comment:
Nice looks like this catches those namespaced imports like
https://github.com/mongodb-js/compass/blob/2762bd08a9f93d04f01be923e5958eed08ba28c5/packages/databases-collections/src/stores/databases-store.js#L1
👍